### DISCUSSION TITLE
Cannot use Cohere as prompt driver.

### DISCUSSION THREAD

**Discussion by TheDud0407** [griptape-ai/griptape#543] [Q&A] [answered]
Hello everyone!

I am trying to use cohere as prompt driver. I installed the extra but when I run the example is not working: 

import os
from griptape.structures import Agent
from griptape.drivers import CoherePromptDriver

agent = Agent(
    prompt_driver=CoherePromptDriver(
        model="command",
        api_key=os.environ['COHERE_API_KEY'],
    )
)

agent.run('What is the sentiment of this review? Review: "I really enjoyed this movie!"')

Not sure what's going on, is there anything I miss? Using OpenAI is a must?

I am getting this error: openai.OpenAIError: The api_key client option must be set either by passing api_key to the client or by setting the OPENAI_API_KEY environment variable.

**Comment by SavagePencil** [comment, marked as answer]
Definitely a gap and we're working to address it in a more comprehensive manner. You can see a similar discussion [here.](https://github.com/griptape-ai/griptape/issues/509). Until the comprehensive fix appears, you can follow the workaround detailed in that thread. It involves disabling the TaskMemory for the prompt driver, as detailed [here](https://docs.griptape.ai/latest/griptape-framework/structures/prompt-drivers/#prompt-drivers). Will leave this open until we get the fix out.
